input/input.c

```diff
@@ -604,8 +616,84 @@ static void interpret_key(struct input_ctx *ictx, int code, double scale)
 
     memset(ictx->key_history, 0, sizeof(ictx->key_history));
 
-    cmd->scale = scale;
-    mp_input_queue_cmd(ictx, cmd);
+    if (mp_input_is_scalable_cmd(cmd)) {
+        cmd->scale = scale;
+        mp_input_queue_cmd(ictx, cmd);
+    } else {
+        // Non-scalable commands won't understand cmd->scale, so synthesize
+        // multiple commands with cmd->scale = 1
+        cmd->scale = 1;
+        for (int i = 0; i < scale_units - 1; i++)
```

Comment:
Yeah, makes sense. I saw values of 4 or 5 during testing. Maybe misbehaving mouse drivers could send even higher deltas in a single event? A limit of 20 could work.